packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTableView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _cellDecorator = _createCellDecorator();
+  }
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    _cellDecorator = _createCellDecorator();
+  }
+
+  IndexedWidgetBuilder _createCellDecorator() {
+    if (widget.plainChildren != null) {
+      return (BuildContext context, int index) {
+        if (index >= widget.plainChildren.length) {
+          return null;
+        }
+
+        final bool isLastCell = index == widget.plainChildren.length - 1;
+        return _decorateCell(widget.plainChildren[index], isLastCell);
+      };
+    } else if (widget.plainChildrenBuilder != null) {
+      return (BuildContext context, int index) {
+        final Widget cell = widget.plainChildrenBuilder(context, index);
+        if (cell == null) {
+          return null;
+        }
+
+        bool isLastCell;
+        if (widget.plainChildCount != null) {
+          // The child count was provided so we will simply compare cell indices.
+          isLastCell = index == widget.plainChildCount - 1;
+        } else {
+          // No child count was provided, so the only way to know if this is the
+          // last cell is to build the next cell and check for null.
+          final Widget nextCell = widget.plainChildrenBuilder(context, index + 1);
+          isLastCell =  nextCell == null;
+        }
+
+        return _decorateCell(cell, isLastCell);
+      };
+    } else {
+      throw Exception('No children found.');
+    }
+  }
+
+  // Decorates an individual cell with a divider beneath it. If this cell
+  // [isLastCell] in the table then this decorator also adds a
+  // [_LayoutSizeReport] so that the height of this last cell can be reported
+  // to the extra dividers sliver.
+  Widget _decorateCell(Widget cell, bool isLastCell) {
+    final Widget cellWithDivider = _CupertinoTableViewDivider(
+      child: cell,
+    );
+
+    if (isLastCell) {
+      return _LayoutSizeReport(
+        sizeCallback: (Size size) {
+          if (size != _lastItemSize) {
+            WidgetsBinding.instance.addPostFrameCallback((Duration dt) {
+              setState(() {
+                _lastItemSize = size;
+              });
+            });
+          }
+        },
+        child: cellWithDivider,
+      );
+    } else {
+      return cellWithDivider;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Visual list of cells.
+    final Widget sliverList = SliverList(
+      delegate: SliverChildBuilderDelegate(
+        _cellDecorator,
+        childCount: widget.plainChildCount,
+      ),
+    );
+
+    // Visual list of cells + extra dividers at the bottom.
+    final Widget scrollingList = CustomScrollView(
+      controller: widget.scrollController,
+      slivers: <Widget>[]
+        ..add(sliverList)
+        ..add(CupertinoTableViewExtraDividers(
```

Comment:
(didn't read the rest yet) does this class need to be public?

Reply:
for testing